### PR TITLE
WRN-1372: Migrated unit tests for Item to Testing Library

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated unit tests from Enzyme to Testing Library for Item component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-1365

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: Item/tests/Item-specs.js [hunk 1]
```diff
@@ -1,147 +1,106 @@
...
-			<ItemBase label={expected}>
-				Hello Item
-			</ItemBase>
-		);
-
-		const actual = subject.find(`.${css.label}`).first().text();
+		const {getByTestId} = render(<ItemBase data-testid="item" label={expected}>Hello Item</ItemBase>);
+		const actual = getByTestId('item').children.item(1).children.item(0).children.item(1).textContent;
 
 		expect(actual).toBe(expected);
```
Comment (adrian-cocoara-lgp) [inline Item/tests/Item-specs.js:24]:
```suggestion
		test('should support adding a `label`', () => {
		const expected = 'Example Label';

		const {getByText} = render(<ItemBase label={expected}>Hello Item</ItemBase>);
		const label = getByText('Example Label');

		expect(label).toBeInTheDocument();
	});
```
In my opinion, this is easier to read

File: Item/tests/Item-specs.js [hunk 2]
```diff
@@ -1,147 +1,106 @@
...
-
 		const expected = '0';
-		const actual = subject.find(`.${css.label}`).first().text();
 
-		expect(actual).toContain(expected);
+		const {getByTestId} = render(<ItemBase data-testid="item" label={expected}>Hello Item</ItemBase>);
+		const actual = getByTestId('item').children.item(1).children.item(0).children.item(1).textContent;
+
+		expect(actual).toBe(expected);
 	});
```
Comment (adrian-cocoara-lgp) [inline Item/tests/Item-specs.js:34]:
```suggestion
	test('should support label with 0', () => {
		const expected = '0';

		const {getByText} = render(<ItemBase label={expected}>Hello Item</ItemBase>);
		const label = getByText('0');

		expect(label).toBeInTheDocument();
	});
```
Same reason as above

File: Item/tests/Item-specs.js [hunk 3]
```diff
@@ -1,147 +1,106 @@
...
-				{expected}
-			</ItemBase>
-		);
-
-		const actual = subject.find(`.${css.content}`).first().text();
+		const {getByTestId} = render(<ItemBase data-testid="item" label="Example Label">{expected}</ItemBase>);
+		const actual = getByTestId('item').children.item(1).children.item(0).children.item(0).textContent;
 
 		expect(actual).toBe(expected);
 	});
```
Comment (adrian-cocoara-lgp) [inline Item/tests/Item-specs.js:43]:
```suggestion
	test('should support adding text as a child when a label is also set', () => {
		const expected = 'Hello Item';

		const {getByText} = render(<ItemBase label="Example Label">{expected}</ItemBase>);
		const actual = getByText('Hello Item');

		expect(actual).toBeInTheDocument();
	});
```

File: Item/tests/Item-specs.js [hunk 4]
```diff
@@ -1,147 +1,106 @@
...
-				Hello Item
-			</ItemBase>
-		);
-
-		const actual = subject.find(`.${css.slotBefore}`).first().text();
+		const {getByTestId} = render(<ItemBase data-testid="item" slotBefore={expected}>Hello Item</ItemBase>);
+		const actual = getByTestId('item').children.item(1).textContent;
 
 		expect(actual).toBe(expected);
 	});
```
Comment (adrian-cocoara-lgp) [inline Item/tests/Item-specs.js:52]:
```suggestion
	test('should support `slotBefore`', () => {
		const expected = 'slot before';

		const {getByText} = render(<ItemBase slotBefore={expected}>Hello Item</ItemBase>);
		const actual = getByText('slot before');

		expect(actual).toBeInTheDocument();
	});
```

File: Item/tests/Item-specs.js [hunk 5]
```diff
@@ -1,147 +1,106 @@
...
-			</ItemBase>
-		);
-
-		const actual = subject.find(`.${css.slotAfter}`).first().text();
+		const {getByTestId} = render(<ItemBase data-testid="item" slotAfter={expected}>Hello Item</ItemBase>);
+		const actual = getByTestId('item').children.item(2).textContent;
 
 		expect(actual).toBe(expected);
+
 	});
```
Comment (adrian-cocoara-lgp) [inline Item/tests/Item-specs.js:62]:
```suggestion
	test('should support `slotAfter`', () => {
		const expected = 'slot after';

		const {getByText} = render(<ItemBase slotAfter={expected}>Hello Item</ItemBase>);
		const actual = getByText('slot after');

		expect(actual).toBeInTheDocument();

	});
```

File: Item/tests/Item-specs.js [hunk 6]
```diff
@@ -1,147 +1,106 @@
...
-				Hello Item
-			</ItemBase>
-		);
-
-		const actual = subject.find(`.${css.slotAfter}`).first().text();
+		const {getByTestId} = render(<ItemBase data-testid="item" slotAfter={expected}>Hello Item</ItemBase>);
+		const actual = getByTestId('item').children.item(2).textContent;
 
 		expect(actual).toBe(expected);
+
```
Comment (alexandrumorariu) [inline Item/tests/Item-specs.js:61]: Please remove this space

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Pull request merged**